presto-hive/src/main/java/io/prestosql/plugin/hive/HiveLocationService.java

```diff
@@ -53,7 +53,7 @@ public HiveLocationService(HdfsEnvironment hdfsEnvironment)
     }
 
     @Override
-    public LocationHandle forNewTable(SemiTransactionalHiveMetastore metastore, ConnectorSession session, String schemaName, String tableName, Optional<Path> externalLocation)
+    public LocationHandle forNewTable(SemiTransactionalHiveMetastore metastore, ConnectorSession session, String schemaName, String tableName, Optional<Path> externalLocation, boolean withData)
```

Comment:
It is not the first time when "WITH NO DATA" is overlooked and causes problems.
@martint Why isn't `CREATE TABLE AS ... WITH NO DATA` simply translated into `CREATE TABLE` for connector's sake?

Reply:
Isn't it translated like that? Hence this PR with change in `createTable`?